doc/designs/quic-design/quic-api.md

```diff
@@ -0,0 +1,1659 @@
+      - [`SSL_stream_conclude`](#-ssl-stream-conclude-)
+      - [`SSL_stream_reset`](#-ssl-stream-reset-)
+      - [`SSL_get_stream_state`](#-ssl-get-stream-state-)
+      - [`SSL_get_stream_read_error_code`, `SSL_get_stream_write_error_code`](#-ssl-get-stream-read-error-code----ssl-get-stream-write-error-code-)
+      - [`SSL_get_conn_close_info`](#-ssl-get-conn-close-info-)
+    + [New APIs for Multi-Stream Operation](#new-apis-for-multi-stream-operation)
+      - [Notes on Multi-Threaded Operation](#notes-on-multi-threaded-operation)
+      - [Notes on Blocking](#notes-on-blocking)
+      - [Notes on Application-Level Polling](#notes-on-application-level-polling)
+      - [`SSL_get0_connection`](#-ssl-get0-connection-)
+      - [`SSL_is_connection`](#-ssl-is-connection-)
+      - [`SSL_get_stream_type`](#-ssl-get-stream-type-)
+      - [`SSL_get_stream_id`](#-ssl-get-stream-id-)
+      - [`SSL_new_stream`](#-ssl-new-stream-)
+      - [`SSL_accept_stream`](#-ssl-accept-stream-)
+      - [`SSL_get_accept_stream_queue_len`](#-ssl-get-accept-stream-queue-len-)
+      - [`SSL_set_incoming_stream_policy`](#-ssl-set-incoming-stream-policy-)
+      - [`SSL_set_default_stream_mode`](#-ssl-set-default-stream-mode-)
+    + [Future APIs](#future-apis)
+  * [BIO Objects](#bio-objects)
+    + [Existing APIs](#existing-apis-1)
+      - [`BIO_s_connect`, `BIO_new_ssl_connect`, `BIO_set_conn_hostname`](#-bio-s-connect----bio-new-ssl-connect----bio-set-conn-hostname-)
+      - [`BIO_new_bio_pair`](#-bio-new-bio-pair-)
+      - [Interactions with `BIO_f_buffer`](#interactions-with--bio-f-buffer-)
+      - [MTU Signalling](#mtu-signalling)
+    + [New APIs](#new-apis)
+      - [`BIO_sendmmsg` and `BIO_recvmmsg`](#-bio-sendmmsg--and--bio-recvmmsg-)
+      - [Truncation Mode](#truncation-mode)
+      - [Capability Negotiation](#capability-negotiation)
+      - [Local Address Support](#local-address-support)
+      - [`BIO_s_dgram_pair`](#-bio-s-dgram-pair-)
+      - [`BIO_POLL_DESCRIPTOR`](#-bio-poll-descriptor-)
+      - [`BIO_s_dgram_mem`](#-bio-s-dgram-mem-)
+      - [`BIO_err_is_non_fatal`](#-bio-err-is-non-fatal-)
+  * [Q & A](#q---a)
+
+Overview and Implementation Status
+----------------------------------
+
+A listing of all SSL object APIs and their implications for QUIC, including
+current implementation status, can be found in
+[quic-api-ssl-funcs.md](./quic-api-ssl-funcs.md).
+
+Non-SSL object APIs which are new or changed, or otherwise discussed in this
+document are listed below, along with their implementation status. SSL object
+APIs are not listed here; see [quic-api-ssl-funcs.md](./quic-api-ssl-funcs.md)
+for details on SSL object APIs.
+
+| Semantics | API                             | Status |
+|-----------|---------------------------------|--------|
+| TBD       | `BIO_s_connect`                 | TODO  |
+| TBD       | `BIO_set_conn_hostname`         | TODO   |
+| TBD       | `BIO_new_bio_pair`              | TODO   |
```

Comment:
This seems out of date

Reply:
Almost. I need to do a final review on our requirements here. I'd leave this for now.